src/PSR7/Exception/InvalidMessage.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace League\OpenAPIValidation\PSR7\Exception;
+
+use Throwable;
+
+class InvalidMessage extends ValidationFailed
+{
+    public const MESSAGE_SERVER_REQUEST = 0;
+    public const MESSAGE_REQUEST        = 1;
+    public const MESSAGE_RESPONSE       = 2;
+
+    public static function becauseServerRequestInvalid(Throwable $e) : self
+    {
+        return new static('Request message failed validation', self::MESSAGE_SERVER_REQUEST, $e);
+    }
+
+    public static function becauseRequestInvalid(Throwable $e) : self
+    {
+        return new static('Request message failed validation', self::MESSAGE_REQUEST, $e);
+    }
+
+    public static function becauseResponseInvalid(Throwable $e) : self
+    {
+        return new static('Request message failed validation', self::MESSAGE_RESPONSE, $e);
+    }
+
+    public function isResponse() : bool
+    {
+        return $this->code === self::MESSAGE_RESPONSE;
+    }
+
+    public function isRequest() : bool
+    {
+        return $this->code === self::MESSAGE_REQUEST;
+    }
+
+    public function isServerRequest() : bool
```

Comment:
I feel previous approach with separate exceptions way more extensible since anyone can write own validator and add own exception similar to existing. with this checks I don't see it's possible

Reply:
They can still extend ValidationFailed

Reply:
Yeap, it makes sense! let me revert it